Source/WebGPU/WebGPU/RenderPipeline.mm

```diff
@@ -32,10 +32,258 @@
+    // https://gpuweb.github.io/gpuweb/#abstract-opdef-validating-gpurenderpipelinedescriptor
+
+    // FIXME: we additionally reject valid descriptors using unimplemented features.
+    // Remove these checks once we support them.
+
+    // Does not support module constants in vertex shaders
+    if (descriptor.vertex.constantCount)
+        return false;
+    // Does not support vertex shader input buffers
+    if (descriptor.vertex.bufferCount)
+        return false;
+
+    if (descriptor.fragment) {
+        const auto& fragmentDescriptor = *descriptor.fragment;
+
+        // Does not support module constants in fragment shaders
+        if (fragmentDescriptor.constantCount)
+            return false;
+    }
+
+    // Does not support depth stencils.
+    if (descriptor.depthStencil)
+        return false;
+
+    // Does not support multisampling
+    if (descriptor.multisample.count > 1)
+        return false;
+    if (descriptor.multisample.alphaToCoverageEnabled)
+        return false;
+
+    return true;
+}
+
 Ref<RenderPipeline> Device::createRenderPipeline(const WGPURenderPipelineDescriptor& descriptor)
 {
-    UNUSED_PARAM(descriptor);
-    return RenderPipeline::createInvalid(*this);
+    if (!validateRenderPipeline(descriptor))
+        return RenderPipeline::createInvalid(*this);
+
+    MTLRenderPipelineDescriptor* mtlRenderPipelineDescriptor = [MTLRenderPipelineDescriptor new];
+
+    {
+        if (descriptor.vertex.nextInChain)
+            return RenderPipeline::createInvalid(*this);
+
+        const auto& vertexModule = WebGPU::fromAPI(descriptor.vertex.module);
+        const auto& vertexFunctionName = String::fromLatin1(descriptor.vertex.entryPoint);
+
+        const auto vertexFunction = vertexModule.getNamedFunction(vertexFunctionName);
+        if (!vertexFunction)
+            return RenderPipeline::createInvalid(*this);
+
+        mtlRenderPipelineDescriptor.vertexFunction = vertexFunction;
+    }
+
+    if (descriptor.fragment) {
+        const auto& fragmentDescriptor = *descriptor.fragment;
+
+        if (fragmentDescriptor.nextInChain)
+            return RenderPipeline::createInvalid(*this);
+
+        const auto& fragmentModule = WebGPU::fromAPI(fragmentDescriptor.module);
+        const auto& fragmentFunctionName = String::fromLatin1(fragmentDescriptor.entryPoint);
+
+        const auto fragmentFunction = fragmentModule.getNamedFunction(fragmentFunctionName);
+        if (!fragmentFunction)
+            return RenderPipeline::createInvalid(*this);
+
+        mtlRenderPipelineDescriptor.fragmentFunction = fragmentFunction;
+
+        for (uint32_t i = 0; i < fragmentDescriptor.targetCount; ++i) {
+            const auto& targetDescriptor = fragmentDescriptor.targets[i];
+            const auto& mtlColorAttachment = mtlRenderPipelineDescriptor.colorAttachments[i];
+
+            mtlColorAttachment.pixelFormat = Texture::pixelFormat(targetDescriptor.format);
```

Comment:
`static MTLPixelFormat pixelFormat` in `Texture.mm` should be exported to `static Texture::pixelFormat` so the render pipeline creation routine can reuse it to convert the WGSL color attachment pixel format to Metal.